src/xma/src/xmaapi/xma_utils.cpp

```diff
@@ -109,22 +109,15 @@ namespace xma_core {
     }
 
     int32_t create_session_execbo(XmaHwSessionPrivate *priv, int32_t count, const std::string& prefix) {
-        for (int32_t d = 0; d < count; d++) {
-            xclBufferHandle  bo_handle = 0;
-            int       execBO_size = MAX_EXECBO_BUFF_SIZE;
-            char     *bo_data;
-            bo_handle = priv->dev_handle.get_handle()->alloc_bo(execBO_size, XCL_BO_FLAGS_EXECBUF);
-            if (!bo_handle || bo_handle == NULLBO)
-            {
-                xma_logmsg(XMA_ERROR_LOG, prefix.c_str(), "Initalization of plugin failed. Failed to alloc execbo");
-                return XMA_ERROR;
-            }
-            bo_data = reinterpret_cast<char*>(priv->dev_handle.get_handle()->map_bo(bo_handle, true));
-            memset((void*)bo_data, 0x0, execBO_size);
+        for (int32_t d = 0; d < count; d++) {  
             priv->kernel_execbos.emplace_back(XmaHwExecBO{});
             XmaHwExecBO& dev_execbo = priv->kernel_execbos.back();
-            dev_execbo.handle = bo_handle;
-            dev_execbo.data = bo_data;
+            std::string cu_name = std::string(reinterpret_cast<char*>(priv->kernel_info->name));
+            int pos = cu_name.find(":");
```

Comment:
Just curious: what happens if no `:` is found?

Reply:
handling error check now and submitted these changes to the PR.